py/server/deephaven/experimental/iceberg.py

```
@@ -271,9 +272,95 @@ def from_current_snapshot(cls) -> 'SchemaProvider':
+        """
+        return cls(_JSortOrderProvider.unsorted())
+
+    @classmethod
+    def use_table_default(cls) -> 'SortOrderProvider':
+        """
+        Use the default sort order of the table while writing new data. If no sort order is set on the table, no sorting
+        will be done.
+
+        Returns:
+            the `SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.useTableDefault())
+
+    @classmethod
+    def from_sort_id(cls, sort_order_id: int) -> 'SortOrderProvider':
+        """
+        Use the sort order with the given ID to sort new data while writing to the Iceberg table.
+
+        Args:
+            sort_order_id (int): the id of the sort order to use.
+
+        Returns:
+            the `.SortOrderProvider` object.
+        """
+        return cls(_JSortOrderProvider.fromSortId(sort_order_id))
+
+    def with_id(self, sort_order_id: int) -> 'SortOrderProvider':
+        """
+        Returns a sort order provider that uses the current provider to determine the columns to sort on, but writes a
+        different sort order ID to the Iceberg table.
+        For example, this provider might sort by columns {A, B, C}, but the ID written to Iceberg corresponds to a sort
+        order with columns {A, B}.
+
+        Args:
+            sort_order_id (int): the sort order ID to write to the Iceberg table.
+
+        Returns:
+            the `SortOrderProvider` object.
+        """
+        return SortOrderProvider(self._j_object.withId(sort_order_id))
+
+    def with_fail_on_unmapped(self, fail_on_unmapped: bool) -> 'SortOrderProvider':
+        """
+        Returns a sort order provider that will fail if the sort order cannot be applied to the tables being written.
+        By default, if the sort order cannot be applied, the tables will be written without sorting.
+
+        Args:
+            fail_on_unmapped: whether to fail if the sort order cannot be applied to the tables being written
```

This function gives me a lot of confusion.
1. It talks about default behavior, but it does not have a default value for `fail_on_unmapped`.
2. Defaulting to not failing seems weird, instead of defaulting to failing.
3. When it "fails", it still charges ahead doing something.  This doesn't really sound like "failure" to me.

Overall, the behavior and documentation of this method don't make sense to me.

---

Made the change in the behaviour and the documentation. Hopefully this will be more clear.